packages/lodestar/src/chain/forkChoice/arrayDag/lmdGhost.ts

```diff
@@ -98,18 +98,21 @@ export class Node {
     return typeof(this.parent) === "number" && this.parent >= 0;
   }
 
-  public shiftIndex(n: number): void {
-    assert.gte(n, 0, "Shift index must be positive");
+  public shiftIndex(oldNodes: Node[], newNodeIndexes: Map<RootHex, number>): void {
     if (this.hasBestChild()) {
-      this.bestChild = this.bestChild - n;
+      const bestChildRoot = oldNodes[this.bestChild].blockRoot;
+      this.bestChild = newNodeIndexes.get(bestChildRoot);
     }
     if (this.hasParent()) {
-      this.parent = this.parent - n;
+      const parentRoot = oldNodes[this.parent].blockRoot;
+      // this may be undefined because its parent is deleted from node
+      this.parent = newNodeIndexes.get(parentRoot);
```

Comment:
There should only be one node w/o a parent, the finalized node, right?

Reply:
yeah I think so